packages/gitbook-docs/src/markdoc/tokenProcessor.ts

```diff
@@ -29,12 +29,19 @@ const makeHtmlParser = () => {
       result.push(token);
     },
     ontext: (content) => {
-      if (typeof content === 'string' && content.trim().length > 0) {
+      if (typeof content !== 'string') return;
+      if (content.trim().length > 0) {
```

Comment:
What about to replace the `else if` with the following condition?
```suggestion
      if (content.trim().length > 0 || content === ' ') {
```